p2p/p2p.go

```diff
@@ -0,0 +1,69 @@
+package p2p
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.nanomsg.org/mangos/v3/protocol/bus"
+	_ "go.nanomsg.org/mangos/v3/transport/tcp"
+	"go.uber.org/zap"
+
+	"github.com/Zamerykanizowana/replicated-file-system/config"
+)
+
+const transportScheme = "tcp"
+
+type peer struct {
+	Address string
+	Name    string
+}
+
+func Run(selfConfig *config.PeerConfig, peersConfig []*config.PeerConfig) error {
+	sock, err := bus.NewSocket()
+	if err != nil {
+		zap.L().With(zap.Error(err)).Fatal("failed to setup BUS protocol socket")
+	}
+	selfURL := buildURL(selfConfig)
+	if err = sock.Listen(selfURL.String()); err != nil {
+		zap.L().With(zap.Error(err)).Fatal("failed to start listening on the BUS")
+	}
+
+	peers := make([]peer, 0, len(peersConfig))
+	for _, p := range peersConfig {
+		peerURL := buildURL(p)
+		peers = append(peers, peer{
+			Address: peerURL.String(),
+			Name:    p.Name,
+		})
+	}
+
+	time.Sleep(time.Second)
+
+	for _, p := range peers {
+		if err = sock.Dial(p.Address); err != nil {
+			return err
+		}
+	}
+	time.Sleep(time.Second)
+
+	zap.L().Info(fmt.Sprintf("%s: SENDING '%s' ONTO BUS\n", selfConfig.Name, selfConfig.Name))
+	if err = sock.Send([]byte(selfConfig.Name)); err != nil {
+		return err
+	}
+	for range peersConfig {
+		msg, err := sock.Recv()
```

Comment:
...and same here regarding `errors.Wrap`. Actually it's more of an open-ended question on how do we decide whether to return an error directly or wrap it.

Reply:
refer to my comment above :)